### PR TITLE
Search

### PR DESCRIPTION
**Pull request by michaeleggboy**
## Change Log
- Users can now filter/search for bookmarks/clinks. An autocomplete dropdown opens up as users type, on select it filters to the specific bookmark/clink. User can also press enter or press the search button to filter bookmarks/clinks by search parameter. 

- Deleted bookmarkcard.js, as wasn't in use.

## Screenshots
|  Before 	|   After	|
|---	|---	|
|  image|image|
| image| image|

## Next Steps
- Use custom urls to grab information, rather than 'aesthetically' as we are doing now.
- Make sure response objects and api are updated with database changes.

### CODE REVIEW

**Comment by clairezc27** [comment]
Great work! When I type in the search bar for the current clink I'm viewing and then switch clinks, the search bar doesn't clear itself. Did we want to clear the text, or to keep the text in the search bar?

**Comment by michaeleggboy** [comment]
> Great work! When I type in the search bar for the current clink I'm viewing and then switch clinks, the search bar doesn't clear itself. Did we want to clear the text, or to keep the text in the search bar?

I'm not sure that you can just auto clear the search bar but the search parameter is automatically reset when you change clink or user page, but I will look into clearing.

File: react-app/src/App.js [hunk 1]
```diff
@@ -13,11 +13,10 @@ function App() {
     <Router>
       <div>
         <Switch>
-          <Route exact path="/get-started/:form" component={GetStartedPage} />
-          <Route exact path="/dashboard" component={Dashboard} />
-          <Route exact path="/dashboard/:clink" component={Dashboard} />
+          <Route exact path="/get-started/:form?" component={GetStartedPage} />
+          <Route exact path="/dashboard/:clinkId?/:searchParams?" component={Dashboard} />
```
Comment (marsw42) [inline react-app/src/App.js:17]: Do not use a subpath for search, use a query param instead
https://learnwithparam.com/blog/how-to-handle-query-params-in-react-router/

**Comment by michaeleggboy** [comment]
Updating ...

**Pull request merged**